includes/Traits/FieldableTrait.php

```diff
@@ -439,101 +453,207 @@ public function update_post_meta( $meta_vars, $post_id ) {
+                            }
+                        } else {
+                            $existing_term = term_exists( $posted_terms, 'product_cat' );
+                            if ( $existing_term ) {
+                                $term_ids[] = $existing_term['term_id'];
+                            } else {
+                                $new_term = wp_insert_term( $posted_terms, 'product_cat' );
+                                if ( !is_wp_error($new_term) ) {
+                                    $term_ids[] = $new_term['term_id'];
+                                }
                             }
+                        }
+                    }
+                }
+                wp_set_object_terms( $post_id, $term_ids, 'product_cat' );
+                continue;
+            }
 
-                            $new_term = wp_insert_term( $term_name, $taxonomy['name'] );
+            // Handle product tags if WooCommerce is enabled
+            if ( $is_woo_taxonomy && 'product_tag' === $taxonomy_name ) {
+                if ( empty( $posted_terms ) ) {
+                    wp_set_object_terms( $post_id, [], 'product_tag' );
+                    continue;
+                }
 
-                            return $new_term['term_id'];
-                        }, $terms
-                    );
+                $terms_to_set = [];
+                $terms_array = is_array( $posted_terms ) ? $posted_terms : [ $posted_terms ];
+
+                foreach ( $terms_array as $term ) {
+                    if ( is_numeric( $term ) ) {
+                        // If ID is provided, get the term name
+                        $term_obj = get_term( $term, 'product_tag' );
+                        if ( $term_obj && ! is_wp_error( $term_obj ) ) {
+                            $terms_to_set[] = $term_obj->name;
+                        }
+                    } else {
+                        // If name is provided, use it directly
+                        $terms_to_set[] = $term;
+                    }
+                }
 
-                    $taxonomy_name = array_filter( $terms );
+                if ( ! empty( $terms_to_set ) ) {
+                    wp_set_post_terms( $post_id, $terms_to_set, 'product_tag', false );
                 }
+                continue;
             }
 
-            // At this point $taxonomy_name should be a single id or array of ids
-            if ( isset( $taxonomy_name ) && $taxonomy_name != 0 && $taxonomy_name != -1 ) {
-                if ( is_object_in_taxonomy( $this->form_settings['post_type'], $taxonomy['name'] ) ) {
-                    $tax = $taxonomy_name;
-                    // if it's not an array, make it one
-                    if ( ! is_array( $tax ) ) {
-                        $tax = [ $tax ];
+            // Handle other WooCommerce taxonomies if WooCommerce is enabled
+            if ( $is_woo_taxonomy && (
+                'pa_' === substr( $taxonomy_name, 0, 3 ) || 
```

Comment:
do not check this way. use ` This taxonomy is a WooCommerce attribute` option from form builder field setting